int32 pin_id
string name
bool state
